src/libsyntax/parse/parser.rs

```diff
@@ -4699,7 +4699,7 @@ impl<'a> Parser<'a> {
                     SeqSep::trailing_allowed(token::Comma),
                     |p| p.parse_fn_block_arg()
                 );
-                self.bump();
+                self.expect(&token::BinOp(token::Or))?;
```

Comment:
Hmm, `token::OrOr` should be possible here as well in theory, e.g. 
```rust
#![allow(unused)]

fn main() {
    let x = |x, y| |a, b| 0; // Legal
    x(1, 2)(3, 4);
}
```
, but `|x, y||a, b| 0` (without space) doesn't parse on stable/beta/nightly already.
Worth fixing as well!

Reply:
I'm trying to fix this. Should I work on this double-closure parsing bug in a new PR?